server/src/main/java/org/elasticsearch/action/ActionModule.java

```diff
@@ -444,17 +441,17 @@ public ActionModule(Settings settings, IndexNameExpressionResolver indexNameExpr
             UnaryOperator<RestHandler> newRestWrapper = plugin.getRestHandlerWrapper(threadPool.getThreadContext());
             if (newRestWrapper != null) {
                 logger.debug("Using REST wrapper from plugin " + plugin.getClass().getName());
+                if (plugin.getClass().getCanonicalName() == null ||
+                    plugin.getClass().getCanonicalName().startsWith("org.elasticsearch.xpack") == false) {
+                    logger.error("The " + plugin.getClass().getName() + " plugin tried to install a custom REST wrapper. This " +
+                        "functionality is not available anymore.");
```

Comment:
Nit: Since an exception is thrown immediately after this and will be logged. It seems this logging is no longer necessary.